### PR TITLE
Add macro benchmarks and upgrade dotty version

### PR DESCRIPTION
**Pull request by nicolasstucki**
Note: this version of dotty used the `rewrite` keyword instead of `inline`
      which will be changed back in the next release.

### CODE REVIEW

**Comment by nicolasstucki** [comment]
@biboudis do not merge this for now. I would like to have it upgraded to the next release version of Dotty to have the `inline` keyword back.

**Comment by nicolasstucki** [comment]
@biboudis now we can merge this. I used a nightly that used `inline` again. could you have a quick look at the changes before merging?

**Review by biboudis — APPROVED** [review]

**Pull request merged**
